src/phoenix/server/api/dataloaders/dataset_example_revisions.py

```diff
@@ -95,4 +96,11 @@ async def _load_fn(self, keys: List[Key]) -> List[Union[Result, ValueError]]:
                 ) in await session.stream(query)
                 if is_valid_version
             }
-        return [results.get(key, ValueError("Could not find revision.")) for key in keys]
+        return [
+            results.get(key, DatasetExampleRevisionException("Could not find revision."))
+            for key in keys
+        ]
+
+
+class DatasetExampleRevisionException(PhoenixGraphQLException):
```

Comment:
Too specific. Errors can be generic like NotFound or BadRequest. These can be used to populate the error extension block